go/pkg/repair.go

```
@@ -9,208 +9,81 @@ import (
+// return an error, but a context cancellation always wins over the warning path.
+func removeStaleLock(ctx context.Context, r pmexec.Runner, path string) error {
 	if err := ctx.Err(); err != nil {
 		return err
 	}
-	if result, err := d.run(ctx, "remove", "--duplicates", "-y"); err != nil {
-		if ctxErr := ctx.Err(); ctxErr != nil {
-			return ctxErr
-		}
-		slog.Warn("dnf remove --duplicates failed", "error", err, "stderr", result.Stderr)
+	if _, err := statFile(path); err != nil {
+		return nil // file doesn't exist — nothing to remove
```

golangci-lint (2.12.2)

[error] error is not nil (line 58) but it returns nil

(nilerr)